### PR TITLE
Initial service_account resorce

### PR DESCRIPTION
**Pull request by dharrisio**

### CODE REVIEW

**Review by smikloska — COMMENTED** [review]
looks good at first 👁️, just nit for doc gen and adding example docs

File: internal/provider/service_account.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+			"name": schema.StringAttribute{
+				Description: "Service account name",
+				Required:    true,
+			},
+			"role_id": schema.StringAttribute{
+				Description: "Role ID",
+				Required:    true,
+			},
+			"credential_key": schema.StringAttribute{
+				Description: "Credential ID",
```
Comment (smikloska) [inline internal/provider/service_account.go:68]:
```suggestion
				Description: "Credential Key",
```

File: internal/provider/service_account.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"credential_secret": schema.StringAttribute{
```
Comment (smikloska) [inline internal/provider/service_account.go:74]: Use case is to create service account and be able reference to use credentials in other resources?

Comment (dharrisio) [reply internal/provider/service_account.go:74]: Yes, that's correct. The secret is only available at create time, so imported service accounts will not have a value for this attribute.

**Review by smikloska — APPROVED** [review]

**Pull request merged**